docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,780 @@
+            input: CreateFooRequest,
+            output: CreateFooResponse,
+        }
+
+        structure CreateFooRequest {
+            @mutability("full")
+            tags: TagList,
+        }
+
+        structure CreateFooResponse {
+            fooId: String,
+        }
+
+        structure FooProperties {
+            @mutability("full")
+            barProperty: String,
+        }
+
+
+The following example defines a CloudFormation resource that marks the derivable
+``immutableSetting`` property as create and read only:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#mutability
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: [FooProperties])
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+        structure FooProperties {
+            @mutability("create-and-read")
+            immutableSetting: Boolean,
+        }
+
+
+The following example defines a CloudFormation resource that marks the derivable
+``updatedAt`` and ``createdAt`` properties as read only:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#mutability
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: [FooProperties])
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            read: GetFoo,
+        }
+
+        @readonly
+        operation GetFoo {
+            input: GetFooRequest,
+            output: GetFooResponse,
+        }
+
+        structure GetFooRequest {
+            @required
+            fooId: String
+        }
+
+        structure GetFooResponse {
+            @mutability("read")
+            updatedAt: Timestamp,
+        }
+
+        structure FooProperties {
+            @mutability("read")
+            createdAt: Timestamp,
+        }
+
+
+The following example defines a CloudFormation resource that marks the derivable
+``secret`` and ``password`` properties as write only:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#mutability
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: [FooProperties])
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            create: CreateFoo,
+        }
+
+        operation CreateFoo {
+            input: CreateFooRequest,
+            output: CreateFooResponse,
+        }
+
+        structure CreateFooRequest {
+            @mutability("write")
+            secret: String,
+        }
+
+        structure CreateFooResponse {
+            fooId: String,
+        }
+
+        structure FooProperties {
+            @mutability("write")
+            password: String,
+        }
+
+Given the following model with property mutability traits applied,
+
+.. tabs::
```

Comment:
This example and several others are pretty big. Can you comment inside the example to give more details as to what's happening and how things will resolve? Alternatively, provide a description property in the computed tables to say why something derived the way it did?

Reply:
Added descriptions to the tables for these two examples, as it was easier to relate them than with in-model comments.